app/Http/Controllers/TeamController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Team;
+use App\User;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Foundation\Auth\AuthenticatesUsers;
+use Illuminate\Http\Request;
+
+class TeamController extends Controller
+{
+	const ROLES = 'enum.user_roles';
+
+	public function __construct() {
+		$this->middleware('auth');
+	}
+
+	/**
+	 * Display a listing of the resource.
+	 *
+	 * @return \Illuminate\Http\Response
+	 */
+	public function index() {
+		//
+	}
+
+	/**
+	 * Show the form for creating a new resource.
+	 *
+	 * @return \Illuminate\Http\Response
+	 */
+	public function create() {
+		$students = User::where('role', config(self::ROLES)['student'])
+			->where('id', '!=', Auth::user()->id)
+			->select('id', 'name')->get();
+
+		return view('teams.create', compact('students'));
+	}
+
+	/**
+	 * Store a newly created resource in storage.
+	 *
+	 * @param  \Illuminate\Http\Request  $request
+	 * @return \Illuminate\Http\Response
+	 */
+	public function store(Request $request) {
+		$attributes = request()->validate([
+			'name' => 'required',
+			'photo' => 'nullable',
+		]);
+
+		$attributes['leader_id'] = Auth::user()->id;
+
+		$team = Team::create($attributes);
+		if (!isset($team)) {
```

Comment:
Non-blocking question: si lo estás creando en la línea anterior, when would this get hit?

Reply:
Agree con Katie, `create` siempre regresa una instancia del modelo (i.e. it's always `set`) entonces si quieres checar si la llamada fue exitosa puedes hacer 

```PHP
if($team->exists) {
    // do some magic...
}
```

Reply:
No sé por qué pensé que regresaba `null` en ese caso. My bad! Gracias por el tip!